### ISSUE TITLE
sql: database is closed errors after bouncing teleport process on proxy server

### ISSUE THREAD

**Issue by Aharic** [gravitational/teleport#5083]
### Description

**What happened**:

Whenever customer bounces teleport service on proxy (in the course of daily log rotation), new web connections fail with "sql: database is closed" errors until all old connections are logged out and closed. It appears that the new connections are e being directed to the proxy process which has shut down but is still waiting for connections to drain, rather than the new process which has been started.

**What you expected to happen**:

Expectation is that new connections should be directed to the newly created proxy process while the old connections slowly drain and age out on the old process.

**How to reproduce it (as minimally and precisely as possible)**:

1. Set-up cluster in AWS with separate auth, proxy, and node servers, and put the proxy and auth services behind NLBs (in my recreate I used the aws/terraform/ha-autoscale-cluster setup to launch a quick environment).
2. Log into the cluster in the browser, open a browser connection to the node, and leave it running.
3. On the proxy side, find the proxy server that is hosting the connection and send a HUP signal to the teleport process on that server: `sudo kill -HUP 'cat /var/run/teleport/teleport.pid'`
4. Go back to the cluster browser and try to open a new browser connection to the node. You should see an error (either cache is closed or database is closed, depending on setup) which will force you to logout of the previous connection first before allowing any new connections to be opened to the node.

### Environment
- Teleport version (use `teleport version`): Customer is running 4.4.5-ent but I can also reproduce on 5.0.0-ent (Teleport Enterprise v5.0.0 git:v5.0.0-0-gac4971801 go1.15.5)
- Tsh version (use `tsh version`): 4.4.5-ent and 5.0.0-ent 
- OS (e.g. from `/etc/os-release`): Amazon Linux

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS

**Relevant Debug Logs If Applicable**

[10xgenomics-database-closed-recreate.log](https://github.com/gravitational/teleport/files/5668013/10xgenomics-database-closed-recreate.log)

gz#2309

**Comment by Aharic** [comment]
Request to have this backported to 4.4.5 as well.

**Comment by mpitt** [comment]
I think this is still affecting 6.1.1. My logs are full of lines like

```
Jun 30 08:39:16 ip-172-31-xx-xx teleport[30142]: WARN [NODE]      Failed to get rotation state: database is closed regular/sshserver.go:749
Jun 30 08:39:21 ip-172-31-xx-xx teleport[30142]: WARN [NODE]      Failed to get rotation state: database is closed regular/sshserver.go:749
Jun 30 08:39:26 ip-172-31-xx-xx teleport[30142]: WARN [NODE]      Failed to get rotation state: database is closed regular/sshserver.go:749
Jun 30 08:39:27 ip-172-31-xx-xx teleport[30142]: ERRO [AUTH]      Failed to perform cert rotation check: cache is closed. auth/auth.go:298
```

It's also causing problems to clients connecting:

```
Jun 30 08:39:09 ip-172-31-xx-xx teleport[30142]: ERRO [PROXY:SER] "proxy2021/06/30 08:39:09 http: TLS handshake error from REDACTED:35394: cache is closed\n" utils/cli.go:304
```

The clients get `remote error: tls: internal error` at various stages of the login.

Full version:

```
Teleport v6.1.1 git:v6.1.1-0-geacd474d8 go1.15.5
```

**Comment by travelton** [comment]
@mpitt Please contact our support team at support@goteleport.com and we will help you troubleshoot this. 👍

**Comment by Aharic** [comment]
@russjones is it possible to get this one re-assigned to a new engineer?

**Comment by espadolini** [comment]
I believe that this is either
- some bug that's no longer in supported versions (v7/v8/v9 at time of writing)
- fixed by one or more of #10706, #11074, #11386
- the same as #6945

Therefore I'm closing this as it's either solved or tracked in #6945.
